crates/cxx-qt-lib/src/core/init.cpp

```diff
@@ -11,169 +11,111 @@
 #include <cxx-qt-lib/qset.h>
 #include <cxx-qt-lib/qvector.h>
 
-static const int register_QHash_i32_QByteArray =
+extern "C" bool
+init_cxx_qt_lib_core()
+{
+  static bool initialized = false;
```

Comment:
did we say this might need to be an atomic bool ?

Reply:
Turns out, there is a C++ standard function that does exactly what we need:
[std::call_once](https://en.cppreference.com/w/cpp/thread/call_once)

I'm shocked and awed by the fact that the C++ stdlib actually has an easy-to-use thing that does exactly what it says. 🤯